added examples of image compression

kmanpearl opened the pull request:
This is a very small PR. I added a jupyter notebook that shows example images at different levels of compression.

kmanpearl committed:
added jupyter notebook showing different compressions

kmanpearl committed:
added jupyter notebook showing different compressions

gwaybio reviewed (changes requested):
Really interesting! A couple notes that you should change prior to me performing a full review:

1. Why delete `process_single_movie.py`?
2. Can you move `process_single_plate.ipynb` to a more specific folder (maybe something called `experimentation` or something like that? Should this folder be a github module? If so, then when number should it get?
3. I'm not sure if `process_single_plate.ipynb` is the best name for the file. You're testing different levels of compression! Can you come up with a different, more descriptive name?
4. When adding a jupyter notebook to github, please always make sure that the notebook is executed sequentially (i.e. the first cell is noted with `[1]`, the second `[2]` and so on). This is to ensure reproducibility, since anything can happen in between non-sequentially executed cells that would be hidden to the reader
5. After you move the file (to appropriate folder and name) please run the following command `jupyter nbconvert --to=script --FilesWriter.build_directory=nbconverted *.ipynb`. This will convert the `.ipynb` file to a more readable (and reviewable!) `.py` file.

After you complete these 5 steps, then I will provide an actual code review :)

kmanpearl committed:
made py file from jupyter notebook

kmanpearl commented:
1. I deleted process_single_movie.py since it was basically a prototype of process_multiple_paltes.py. The code from process_sinlge_movie.py is now a part of the image_compresson scripts since testing different compressions is pretty much the only time someone would want to process a single movie.

5. I converted the notebook to py but I did have to make a few changes to make the script run, so they are very very similar but not exactly identical. Not sure if this is a problem.

gwaybio commented:
> I converted the notebook to py but I did have to make a few changes to make the script run, so they are very very similar but not exactly identical. Not sure if this is a problem.

The `.py` file that is nbconverted, does not need to run - it is merely for reviewing purposes. The notebook is what is run.

You can also run the `nbconvert` command over and over again. The notebook and `.py` file should match.

gwaybio commented on experimentation/image_compression.py:
```diff
@@ -0,0 +1,103 @@
+from processor import BasicpyPreprocessor
+import PyBaSiC.pybasic
+
+fiji_path = pathlib.Path("/home/keenanmanpearl/Desktop/mitocheck/fiji-linux64/Fiji.app")
+movie_path = pathlib.Path(
+    "/home/keenanmanpearl/Desktop/mitocheck_movies/movies/LT0001_02/1.ch5/00001_01.ch5"
+)
```
```suggestion
base_dir = pathlib.Path("/home/keenanmanpearl/Desktop")

fiji_path = pathlib.Path(f"{base_dir}/mitocheck/fiji-linux64/Fiji.app")
movie_path = pathlib.Path(
   f"{base_dir}/mitocheck_movies/movies/LT0001_02/1.ch5/00001_01.ch5"
)
```

Please keep in mind [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Please also note that I am making a "suggestion" (the technical term on github) on the `.py` file. Please port all changes to the `.ipynb` file and then run the `nbconvert`

gwaybio commented on experimentation/image_compression.py:
```diff
@@ -0,0 +1,103 @@
+    down_width = int(width / down_factor)
+    down_points = (down_width, down_height)
+    # False for black and white
+    out = cv2.VideoWriter(f"{save_path}", fourcc, fps, down_points, False)
```
instead of position-based function arguments, can you name the arguments?

https://www.pythonmorsels.com/positional-vs-keyword-arguments/

> Named arguments can sometimes make your code a bit more descriptive because you've given a name to an object whose use might not otherwise be clear simply by its position in a function call.

gwaybio replied:
please use named arguments everywhere! It is so helpful for readability :)

gwaybio replied:
(even for functions you yourself write!)

gwaybio commented on experimentation/image_compression.py:
```diff
@@ -0,0 +1,103 @@
+)
+compressed2 = process_movie(2, height, width, save_path, fourcc, fps, 10, corrected_movie)
+
+save_path = pathlib.Path(
```
remember DRY (use the `base_dir` defined previously)

gwaybio commented on experimentation/image_compression.py:
```diff
@@ -0,0 +1,103 @@
+# frames per second
+fps = 4
+
+save_path = pathlib.Path(
```
remember DRY (use the base_dir defined previously)

gwaybio reviewed (approved):
Looks great! Prior to merging please:

1. Confirm nbconverted file is in sync with notebook
2. Use DRY principles
3. Add all named arguments (and you'll likely have to format the function calls, I strongly recommend using `black` for autoformatting jupyter notebooks)

No need for me to take another look! Happy to if it would help

kmanpearl committed:
fixed notebook formatting
